src/QbXml/Objects/Types/DATETIMETYPE.cs

```diff
@@ -156,25 +267,97 @@ public override int GetHashCode()
             return default(DateTime);
         }
 
+
+        public static bool operator <(DATETIMETYPE a, DATETIMETYPE b)
+        {
+            return a.CompareTo(b) < 0;
+        }
+
+        public static bool operator >(DATETIMETYPE a, DATETIMETYPE b)
+        {
+            return a.CompareTo(b) > 0;
+        }
+
         public int CompareTo(DATETIMETYPE other)
         {
-            return this.value.CompareTo(other.value);
+            if (other == null) return 1;
+
+            if (_ignoreOffset || other._ignoreOffset)
+            {
+                return _value.DateTime.CompareTo(other._value.DateTime);
+            }
+
+            return _value.CompareTo(other._value);
         }
 
         public System.Xml.Schema.XmlSchema GetSchema()
         {
             return null;
         }
 
+        [Obsolete]
         public void ReadXml(System.Xml.XmlReader reader)
         {
+            if (!_canReadXml)
+            {
+                //This class needs to be immutable as possible
+                //Only allow this when constructed using the empty constructor to only be used by deserialization
+                throw new InvalidOperationException("This method not to be used by user code");
+            }
+
+            //Only allow ReadXML once to prevent user code from calling this method after deserialized
+            _canReadXml = false;
+
             reader.MoveToContent();
             var isEmptyElement = reader.IsEmptyElement;
             reader.ReadStartElement();
-            if (!isEmptyElement)
+
+            if (isEmptyElement)
+            {
+                _value = new DateTimeOffset(1970, 1, 1, 0, 0, 0, 0, TimeSpan.Zero);
+                _ignoreOffset = true;
+                return;
+            }
+
+            var str = reader.ReadContentAsString();
```

Comment:
shouldn't you move this in your try catch since you have a ReadEndElement in the finally?

Reply:
The try catch is really just for parsing errors. The finally really isn't needed.